shared/constants/types/fs.js

```diff
@@ -2,7 +2,27 @@
 import * as I from 'immutable'
 
 export type _State = {
-  counter: number,
+  path: Path,
+  pathItems: I.Map<Path, PathItem>,
 }
 
 export type State = I.RecordOf<_State>
+
+export type PathItem = {
+  type: PathType,
+  children: I.List<string> | null,
+}
+
+export type Visibility = 'private' | 'public' | 'team' | null
+
+export type PathType = 'folder' | 'file' | 'symlink' | 'exec'
```

Comment:
what is symlink / exec used for. if you want to draw the children in a list as symlinks maybe your list needs to be something other than string. 
aka
```
// children: I.List<string> , instead
children: I.List<ChildType>

type ChildType = {
  type: PathType,
  path: Path
}
```